src/web/app/components/dialogs/FramingModal.tsx

```diff
@@ -74,6 +75,15 @@ const FramingModal = ({ device, onClose, startOnOpen = false }: Props): JSX.Elem
     return () => unregister?.();
   }, []);
 
+  useEffect(() => {
+    if (!promarkModels.has(device.model)) return () => {};
+    const abortPreview = () => manager.current.changeWorkingStatus(false);
+    swiftrayClient.on('disconnected', abortPreview);
+    return () => {
+      swiftrayClient.off('disconnected', abortPreview);
+    };
+  }, [device]);
+
```

Comment:
optional: I thought it would be better to listen the event and change status in framing.ts